### PR TITLE
CDAP-3165 add docs for dbsink

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst [hunk 1]
```diff
@@ -8,16 +8,55 @@ Sinks: Batch: Database
...
+suppose you periodically build a machine learned recommendation model for products
+on your online store. The model is stored in a FileSet and you want to export the contents
+of the FileSet to a database table where it can be served to your users.
 
 .. rubric:: Properties
 
-**columns:** Comma-separated list of columns to export into the specified table.
+**tableName:** Name of the table to export to.
+
+**columns:** Comma-separated list of columns to export to the specified table.
```
Comment (bdmogal) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: Comma-separated list of columns in the specified table to export to.
?

**Comment by bdmogal** [comment]
minor nit, else content LGTM

**Comment by awholegunch** [comment]
hang on: looking

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst [hunk 2]
```diff
@@ -8,16 +8,55 @@ Sinks: Batch: Database
...
 .. rubric:: Description
 
-Batch sink for a database
+Writes records to a database table. Each record will be written to a row in the table.
 
 .. rubric:: Use Case
 
-TODO: Fill me out
+This sink is used whenever you need to write to a database table. For example,
+suppose you periodically build a machine learned recommendation model for products
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: "machine-learned" or maybe "machine learning"

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst [hunk 3]
```diff
@@ -8,16 +8,55 @@ Sinks: Batch: Database
 
 .. rubric:: Description
 
-Batch sink for a database
+Writes records to a database table. Each record will be written to a row in the table.
 
 .. rubric:: Use Case
 
-TODO: Fill me out
+This sink is used whenever you need to write to a database table. For example,
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: I'd drop the "For example", and just start with "Suppose"

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: Are you exporting the list, or exporting the contents of the columns? :)
```
Comma-separated list of columns of the specified table to export.

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst (hunk 2 above)
Comment (albertshau) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: its not really important, maybe i'll just drop the phrase

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst [hunk 4]
```diff
@@ -8,16 +8,55 @@ Sinks: Batch: Database
...
 .. rubric:: Properties
 
-**columns:** Comma-separated list of columns to export into the specified table.
+**tableName:** Name of the table to export to.
+
+**columns:** Comma-separated list of columns to export to the specified table.
+
+**connectionString:** JDBC connection string including database name.
+
+**user:** User identity for connecting to the specified database as. Required for databases that need
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: User identity for connecting to the specified database.

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst [hunk 5]
```diff
@@ -8,16 +8,55 @@ Sinks: Batch: Database
...
+      "columns": "id,name,email,phone",
+      "connectionString": "jdbc:postgresql://localhost:5432/prod",
+      "user": "postgres",
+      "password": "",
+      "jdbcPluginName": "postgres",
+      "jdbcPluginType": "jdbc"
+    }
+  }
+
+This example connects to a database using the specified connectionString, which means
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/database.rst]: specified 'connectionString'

**Comment by awholegunch** [comment]
OK, I'm done.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**